### PR TITLE
feat: dist bucket SSE config

### PR DESCRIPTION
**Pull request by marekaf**
solves #1320 

I took a battle-tested implementation from https://github.com/terraform-aws-modules/terraform-aws-s3-bucket/blob/master/main.tf#L198

example usage:

KMS SSE
```
...

  runner_binaries_s3_sse_configuration = {
    rule = {
      apply_server_side_encryption_by_default = {
        kms_master_key_id = aws_kms_key.dist.arn
        sse_algorithm     = "aws:kms"
      }
    }
  }

...

resource "aws_kms_key" "dist" {
  description             = "KMS key for dist bucket SES"
  deletion_window_in_days = 10
  key_usage               = "ENCRYPT_DECRYPT"
  enable_key_rotation     = true
  is_enabled              = true
  policy                  = data.aws_iam_policy_document.allow_kms.json
}
```

S3 SSE
```
  runner_binaries_s3_sse_configuration = {
    rule = {
      apply_server_side_encryption_by_default = {
        sse_algorithm     = "AES256"
      }
    }
 }
```

the flexibility in implementation is here to allow using either KMS or S3 SSE https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -111,6 +111,12 @@ variable "runner_binaries_syncer_lambda_timeout" {
   default     = 300
 }
 
+variable "runner_binaries_s3_sse_configuration" {
```
[the remark this answers is not in the document]
Comment (npalm) [reply variables.tf:114]: I see no changes since my previous review, did you commit / push the example?

[the remark this answers is not in the document]
Comment (marekaf) [reply variables.tf:114]: https://github.com/philips-labs/terraform-aws-github-runner/pull/1324#issue-1033316759 I put it here. Should I put in inplace in the variables.tf file as a comment?

[the remark this answers is not in the document]
Comment (npalm) [reply variables.tf:114]: The usages of some settings are show in the default example (main.tf) by just adding them as comment. Would that work for you?

[the remark this answers is not in the document]
Comment (marekaf) [reply variables.tf:114]: Sure, I'm on it.

[the remark this answers is not in the document]
Comment (marekaf) [reply variables.tf:114]: done

**Review by npalm — APPROVED** [review]
@marekaf thanks for your contribution, please you rebase? Some conflicts due to merged of other PR (older one).

**Comment by marekaf** [comment]
> @marekaf thanks for your contribution, please you rebase? Some conflicts due to merged of other PR (older one).

rebased and conflicts fixed

it looks like there is a mismatch between contributors' versions (or default configuration) of terraform-docs, somebody keeps switching between `--anchor=true` and `--anchor=false`. It would make sense to add this to `.pre-commit-config.yml`:

```
      - id: terraform_docs
        args:
          - --args=--anchor=true
```

**Comment by npalm** [comment]
> > @marekaf thanks for your contribution, please you rebase? Some conflicts due to merged of other PR (older one).
> 
> rebased and conflicts fixed
> 
> it looks like there is a mismatch between contributors' versions (or default configuration) of terraform-docs, somebody keeps switching between `--anchor=true` and `--anchor=false`. It would make sense to add this to `.pre-commit-config.yml`:
> 
> ```
>       - id: terraform_docs
>         args:
>           - --args=--anchor=true
> ```

CI checks are failing

**Comment by marekaf** [comment]
@npalm oops, sorry. Not sure why pre-commit didn't catch that. Should be fixed now.

**Review by npalm — APPROVED** [review]

**Pull request merged**